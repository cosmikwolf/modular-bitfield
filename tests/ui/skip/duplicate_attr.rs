use modular_bitfield::prelude::*;

#[bitfield]
pub struct TwoExplicitAll {
    #[skip]
    #[skip]
    unused_1: B15,
    a: bool,
}

#[bitfield]
pub struct TwoExplicitListAll {
    #[skip()]
    #[skip()]
    unused_1: B15,
    a: bool,
}

#[bitfield]
pub struct TwoExplicitGetters {
    #[skip(getters)]
    #[skip(getters)]
    unused_1: B15,
    a: bool,
}

#[bitfield]
pub struct TwoExplicitListGetters {
    #[skip(getters, getters)]
    unused_1: B15,
    a: bool,
}

#[bitfield]
pub struct ImplicitExplicitGetters {
    #[skip]
    #[skip(getters)]
    unused_1: B15,
    a: bool,
}

#[bitfield]
pub struct ExplicitImplicitGetters {
    #[skip(getters)]
    #[skip]
    unused_1: B15,
    a: bool,
}

#[bitfield]
pub struct TwoExplicitSetters {
    #[skip(setters)]
    #[skip(setters)]
    unused_1: B15,
    a: bool,
}

#[bitfield]
pub struct TwoExplicitListSetters {
    #[skip(setters, setters)]
    unused_1: B15,
    a: bool,
}

#[bitfield]
pub struct ImplicitExplicitSetters {
    #[skip]
    #[skip(setters)]
    unused_1: B15,
    a: bool,
}

#[bitfield]
pub struct ExplicitImplicitSetters {
    #[skip(setters)]
    #[skip]
    unused_1: B15,
    a: bool,
}

fn main() {}
