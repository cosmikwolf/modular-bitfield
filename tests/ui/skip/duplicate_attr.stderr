error: encountered duplicate `#[skip]` attribute for field
 --> tests/ui/skip/duplicate_attr.rs:6:7
  |
6 |     #[skip]
  |       ^^^^

error: duplicate `#[skip]` here
 --> tests/ui/skip/duplicate_attr.rs:5:7
  |
5 |     #[skip]
  |       ^^^^

error: encountered duplicate `#[skip]` attribute for field
  --> tests/ui/skip/duplicate_attr.rs:14:7
   |
14 |     #[skip()]
   |       ^^^^

error: duplicate `#[skip]` here
  --> tests/ui/skip/duplicate_attr.rs:13:7
   |
13 |     #[skip()]
   |       ^^^^

error: encountered duplicate `#[skip(getters)]` attribute for field
  --> tests/ui/skip/duplicate_attr.rs:22:7
   |
22 |     #[skip(getters)]
   |       ^^^^

error: duplicate `#[skip(getters)]` here
  --> tests/ui/skip/duplicate_attr.rs:21:7
   |
21 |     #[skip(getters)]
   |       ^^^^

error: encountered duplicate `#[skip(getters)]` attribute for field
  --> tests/ui/skip/duplicate_attr.rs:29:21
   |
29 |     #[skip(getters, getters)]
   |                     ^^^^^^^

error: duplicate `#[skip(getters)]` here
  --> tests/ui/skip/duplicate_attr.rs:29:12
   |
29 |     #[skip(getters, getters)]
   |            ^^^^^^^

error: encountered duplicate `#[skip(getters)]` attribute for field
  --> tests/ui/skip/duplicate_attr.rs:37:7
   |
37 |     #[skip(getters)]
   |       ^^^^

error: duplicate `#[skip(getters)]` here
  --> tests/ui/skip/duplicate_attr.rs:36:7
   |
36 |     #[skip]
   |       ^^^^

error: encountered duplicate `#[skip]` attribute for field
  --> tests/ui/skip/duplicate_attr.rs:45:7
   |
45 |     #[skip]
   |       ^^^^

error: duplicate `#[skip]` here
  --> tests/ui/skip/duplicate_attr.rs:44:7
   |
44 |     #[skip(getters)]
   |       ^^^^

error: encountered duplicate `#[skip(setters)]` attribute for field
  --> tests/ui/skip/duplicate_attr.rs:53:7
   |
53 |     #[skip(setters)]
   |       ^^^^

error: duplicate `#[skip(setters)]` here
  --> tests/ui/skip/duplicate_attr.rs:52:7
   |
52 |     #[skip(setters)]
   |       ^^^^

error: encountered duplicate `#[skip(setters)]` attribute for field
  --> tests/ui/skip/duplicate_attr.rs:60:21
   |
60 |     #[skip(setters, setters)]
   |                     ^^^^^^^

error: duplicate `#[skip(setters)]` here
  --> tests/ui/skip/duplicate_attr.rs:60:12
   |
60 |     #[skip(setters, setters)]
   |            ^^^^^^^

error: encountered duplicate `#[skip(setters)]` attribute for field
  --> tests/ui/skip/duplicate_attr.rs:68:7
   |
68 |     #[skip(setters)]
   |       ^^^^

error: duplicate `#[skip(setters)]` here
  --> tests/ui/skip/duplicate_attr.rs:67:7
   |
67 |     #[skip]
   |       ^^^^

error: encountered duplicate `#[skip]` attribute for field
  --> tests/ui/skip/duplicate_attr.rs:76:7
   |
76 |     #[skip]
   |       ^^^^

error: duplicate `#[skip]` here
  --> tests/ui/skip/duplicate_attr.rs:75:7
   |
75 |     #[skip(setters)]
   |       ^^^^
